crates/icrate/examples/browser.rs

```diff
@@ -0,0 +1,284 @@
+type BoxError = Box<dyn std::error::Error + Send + Sync + 'static>;
+type BoxResult<T> = Result<T, BoxError>;
+
+use icrate::{
+    ns_string,
+    objc2::{
+        declare::{Ivar, IvarDrop},
+        declare_class, msg_send,
+        rc::{Id, Shared},
+        runtime::Sel,
+        sel, ClassType, ConformsTo,
+    },
+    AppKit::{
+        NSApplication, NSApplicationActivationPolicyRegular, NSBackingStoreBuffered,
+        NSBezelStyleShadowlessSquare, NSButton, NSColor, NSControl, NSLayoutAttributeHeight,
+        NSLayoutAttributeWidth, NSMenu, NSMenuItem, NSStackView, NSStackViewDistributionFill,
+        NSStackViewDistributionFillEqually, NSTextField, NSTextFieldDelegate, NSTextView,
+        NSUserInterfaceLayoutOrientationHorizontal, NSUserInterfaceLayoutOrientationVertical,
+        NSWindow, NSWindowStyleMaskClosable, NSWindowStyleMaskResizable, NSWindowStyleMaskTitled,
+    },
+    Foundation::{NSObject, NSPoint, NSRect, NSSize, NSURLRequest, NSURL},
+    WebKit::{WKNavigation, WKNavigationDelegate, WKWebView},
+};
+use objc2::{extern_methods, rc::Allocated, runtime::Object};
+
+declare_class! {
+    struct AppDelegate {
+        text_field: IvarDrop<Id<NSTextField, Shared>>,
+        web_view: IvarDrop<Id<WKWebView, Shared>>,
+    }
+
+    unsafe impl ClassType for AppDelegate {
+        type Super = NSObject;
+        const NAME: &'static str = "NavURLDelegate";
+    }
+
+    unsafe impl AppDelegate {
+        #[method(initWithTextField:andWebView:)]
+        #[allow(non_snake_case)]
+        unsafe fn __init_withTextField_andWebView(self: &mut Self, text_field: *mut NSTextField, web_view: *mut WKWebView) -> Option<&mut Self> {
+            let this: Option<&mut Self> = msg_send![super(self), init];
+            this.and_then(|this| {
+                Id::retain(text_field).and_then(|text_field| {
+                    Id::retain(web_view).and_then(|web_view| {
+                        Ivar::write(&mut this.text_field, text_field);
+                        Ivar::write(&mut this.web_view, web_view);
+                        Some(this)
+                    })
+                })
+            })
+        }
+
```

Comment:
Add something similar to `unsafe impl ConformsTo<NSControlTextEditingDelegate> for AppDelegate {` here

Reply:
I actually meant for you to move the `ConformsTo` stuff _inside_ `declare_class!`, since that allows it to ensure that you've implemented any required methods, as well as ensuring that the methods are implemented correctly!